src/rdkafka_msgset_reader.c

```diff
@@ -1216,3 +1401,230 @@ rd_kafka_msgset_parse (rd_kafka_buf_t *rkbuf,
         return err;
 
 }
+
+
+/**
+ * @brief Offset comparator
+ */
+static int rd_kafka_offset_cmp (const void *_a, const void *_b) {
+        const int64_t *a = _a, *b = _b;
+        return *a - *b;
+}
+
+/**
+ * @brief Pid comparator for rd_kafka_aborted_txn_start_offsets_t
+ */
+static int rd_kafka_aborted_txn_cmp_by_pid (const void *_a, const void *_b) {
+        const rd_kafka_aborted_txn_start_offsets_t *a = _a, *b = _b;
+        return a->pid - b->pid;
+}
+
+
+/**
+ * @brief Free resources associated with an AVL tree node.
+ */
+void rd_kafka_aborted_txn_node_destroy (void *offsets) {
+        rd_list_destroy(&((rd_kafka_aborted_txn_start_offsets_t *)offsets)->offsets);
```

Comment:
This cast is hard to read, assign to a temporary variable like we do in cmp

Reply:
also changed the offsets parameter name to `_node_ptr`